src/app/resources/iis_scripts/token_utils.ps1

```diff
@@ -0,0 +1,89 @@
+#Requires -RunAsAdministrator
+#Requires -Version 4.0
+
+Param(
+    [Parameter(Mandatory=$true)]
+    [string]
+    $sessionId,
+
+    [Parameter(Mandatory=$true)]
+    [ValidateSet('ensure', 'delete')]
+    [string]
+    $command,
+
+    [string]
+    $tokenId,
+
+    [string]
+    $apiHost = "https://localhost:55539"
+)
+
+$ErrorActionPreference = "Stop"
+$contentEncoding = [System.Text.Encoding]::UTF8
+$tokenName = "WAC/${sessionId}"
+$RenewEndpoint = "access-tokens"
+$CreateEndpoint = "api-keys"
+$DeleteEndpoint = "api-keys"
+function VerifyResponse([string] $action, [Microsoft.Powershell.Commands.WebResponseObject] $response) {
+    if ($response.StatusCode -ge 300) {
+        throw "Invalid status code $($response.StatusCode) while performing action: $action"
+    }
+}
+function TokenRequest([string] $targetEndpoint, [string]$method, [string]$subpath, $requestBody) {
+    $sessionCreate = Invoke-WebRequest "$apiHost/security/$targetEndpoint" -UseBasicParsing -UseDefaultCredentials -SessionVariable sess

```

Comment:
I think we should catch exception for all the Invoke-WebRequest to get the error information to help users to troubleshoot the problem.

Reply:
Wouldn't the error be thrown upstream?